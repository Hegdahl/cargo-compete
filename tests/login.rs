#![cfg(feature = "__test_with_credentials")]

pub mod common;

use insta::{assert_json_snapshot, assert_snapshot};

#[test]
fn atcoder() -> anyhow::Result<()> {
    let (output, tree) = run("atcoder")?;
    assert_snapshot!("atcoder_output", output);
    assert_json_snapshot!("atcoder_file_tree", tree);
    Ok(())
}

fn run(platform: &str) -> anyhow::Result<(String, serde_json::Value)> {
    common::run(
        |_| Ok(()),
        common::atcoder_credentials()?,
        &["", "compete", "l", platform],
        |_, output| output,
    )
}
