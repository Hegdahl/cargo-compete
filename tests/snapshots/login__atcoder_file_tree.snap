---
source: tests/login.rs
expression: tree
---
{}
